python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+        
+        db = om10.DB(catalog=os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'twinkles_lenses_v2.fits'), vb=False)
+
+        x_offsets = []
+        y_offsets = []
+        total_offsets = []
+
+        for lens_gal_row in spr_agn_lens_df.iterrows():
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_agn_df.query('lens_galaxy_uID == %i' % u_id)
+            use_lens = db.lenses['LENSID'][np.where(db.lenses['twinklesId'] == 
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+            num_img = lens['NIMG']
+
+            for img_idx in range(num_img[0]):
+                # Calculate the offsets from the lens galaxy position
+                offset_x1, offset_y1 = self.offset_on_sky(spr_sys_df['raPhoSim'].iloc[img_idx], 
+                                                          spr_sys_df['decPhoSim'].iloc[img_idx],
+                                                          lens_gal_df['raPhoSim'],
+                                                          lens_gal_df['decPhoSim'])
+
+                x_offsets.append(offset_x1-lens['XIMG'][0][img_idx])
+                y_offsets.append(offset_y1-lens['YIMG'][0][img_idx])
+                total_offsets.append((np.sqrt(x_offsets[-1]**2. + y_offsets[-1]**2.)/
+                                      np.sqrt(lens['XIMG'][0][img_idx]**2. + 
+                                              lens['YIMG'][0][img_idx]**2.)))
+
+        max_total_err = np.max(total_offsets)
+
+        print('------------')
+        print('AGN Location Test Results:')
+
+        if max_total_err < 0.01:
+            print('Pass: Max image offset error less than 1 percent')
+        else:
+            raise CatalogError('\nFail: Max image offset error greater than 1 percent. ' + 
+                               'Max total offset error is: %.4f percent.' % max_total_err)
+
+        print('------------')
+
+        return
+
+    def compare_sne_location(self, spr_sne_df, spr_sne_lens_df):
+        
+        df = pd.read_csv(os.path.join(os.environ['TWINKLES_DIR'], 'data',
+                                      'dc2_sne_cat.csv'))
+
+        x_offsets = []
+        y_offsets = []
+        total_offsets = []
+        images_tested = 0
+
+        for lens_gal_row in spr_sne_lens_df.iterrows():
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_sne_df.query('lens_galaxy_uID == %i' % u_id)
+
+            # There may be no images present in the current instance catalog
+            if len(spr_sys_df) == 0:
+                continue
+
+            # This is just to make sure there are at least some images when we expect
+            # and the previous line is not just skipping everything.
+            images_tested += 1
```

Comment:
Should this be moved into the `for img_idx` loop below, just in case `spr_sys_df['image_number'].values == 0`?